Library/Homebrew/cask/lib/hbc/download_strategy.rb

```diff
@@ -181,7 +181,7 @@ def referer_args
     end
 
     def ext
-      Pathname.new(@url).extname
+      Pathname.new(@url).extname.sub(/\?.*/, '')
```

Comment:
Instead of this, let's use the `@uri_object` from above, to get the path without the query string.

```ruby
@uri_object.path
```

Then, get the extension from this.

```ruby
Pathname.new(@uri_object.path).extname
```